### PR TITLE
Removed `fmt` dependency: was mostly being used for error handling.

### PR DESCRIPTION
**Pull request by ghost**
`fmt` is a big dependency, and I'm looking for ways to remove it in my GopherJS-transpiled code. A lot of packages, including yours, pull `fmt` in just to use `fmt.Errorf`, when `errors.New` would do just as well.

All I've done is change the code to use `errors`. Tests all pass. :)

I may end up using your parser for a JS project I'm writing because the built-in `encoding/json` [uses fmt also](/golang/go/issues/15583) at present!

### CODE REVIEW

File: json.go [hunk 1]
```diff
@@ -235,7 +235,8 @@ func (s *State) getSmallURune(start int) (r rune, err error) {
 	}
 	substr := string(s.data[start+2 : start+6]) // '\u1234'
 	if i, err = strconv.ParseUint(substr, 16, 16); err != nil {
-		err = fmt.Errorf("Could not parse '\\u%v'", substr)
+		err = errors.New("Could not parse '\\u" + substr + "'")
+		//err = fmt.Errorf("Could not parse '\\u%v'", substr)
```
Comment (kurrik) [inline json.go]: Please remove this line

File: json.go [hunk 2]
```diff
@@ -219,7 +219,7 @@ func (s *State) readString() (err error) {
 	s.v = buf.String()
 	if escaped == true {
 		var utfstr = s.v.(string)
-		utfstr = fmt.Sprintf("\"%v\"", utfstr)
+		utfstr = "\"" + utfstr + "\"" // fmt.Sprintf("\"%v\"", utfstr)
```
Comment (kurrik) [inline json.go]: Please remove the trailing comment

File: json.go [hunk 3]
```diff
@@ -477,7 +478,8 @@ func Unmarshal(data []byte, v interface{}) error {
 	)
 	if !svt.AssignableTo(rvt) {
 		if rv.Kind() != reflect.Slice && sv.Kind() != reflect.Slice {
-			return fmt.Errorf("Cannot assign %v to %v", svt, rvt)
+			return errors.New("Cannot assign " + svt.String() + " to " + rvt.String())
+			//return fmt.Errorf("Cannot assign %v to %v", svt, rvt)
```
Comment (kurrik) [inline json.go]: Remove this commented line

File: json.go [hunk 4]
```diff
@@ -491,7 +493,8 @@ func Unmarshal(data []byte, v interface{}) error {
 		)
 		_, ismap = sv.Index(0).Interface().(map[string]interface{})
 		if !(ismap && mapt.AssignableTo(rvte)) {
-			return fmt.Errorf("Cannot assign %v to %v", svte, rvte)
+			return errors.New("Cannot assign " + svte.String() + " to " + rvte.String())
+			//return fmt.Errorf("Cannot assign %v to %v", svte, rvte)
```
Comment (kurrik) [inline json.go]: Remove

**Comment by kurrik** [comment]
I'm open to merging this, would just prefer to have the old lines removed.  Thanks!

**Comment by cathalgarvey** [comment]
Sorry, forgot to remove those before committing, after the tests passed. :)

**Comment by kurrik** [comment]
Merged, thanks!

**Comment by ghost** [comment]
Thanks! :)

**Pull request merged**